src/platform/platform_linux.cpp

```diff
@@ -154,6 +155,69 @@ void update_bridges(std::map<std::string, mp::NetworkInterfaceInfo>& networks)
 }
 } // namespace
 
+std::unique_ptr<QFile> multipass::platform::detail::find_os_release()
+{
+    const std::array<QString, 3> options{QStringLiteral("/var/lib/snapd/hostfs/etc/os-release"),
+                                         QStringLiteral("/var/lib/snapd/hostfs/usr/lib/os-release"),
+                                         QStringLiteral("")};
+
+    auto it = options.cbegin();
+    auto ret = std::make_unique<QFile>(*it);
+    while (++it != options.cend() && !MP_FILEOPS.open(*ret, QIODevice::ReadOnly | QIODevice::Text))
+        ret->setFileName(*it);
+
+    return ret;
+}
+
+std::pair<QString, QString> multipass::platform::detail::parse_os_release(const QStringList& os_data)
+{
+    const QString id_field = "NAME";
+    const QString version_field = "VERSION_ID";
+
+    QString distro_id = "unknown";
+    QString distro_rel = "unknown";
+
+    auto strip_quotes = [](const QString& val) -> QString { return val.mid(1, val.length() - 2); };
+
+    for (const QString& line : os_data)
+    {
+        QStringList split = line.split('=', Qt::KeepEmptyParts);
+        if (split.length() == 2 && split[1].length() > 2) // Check for at least 1 char between quotes.
+        {
+            if (split[0] == id_field)
+                distro_id = strip_quotes(split[1]);
+            else if (split[0] == version_field)
+                distro_rel = strip_quotes(split[1]);
+        }
+    }
+
+    return std::pair(distro_id, distro_rel);
+}
+
+std::string multipass::platform::detail::read_os_release()
+{
+    QFile& os_release = *find_os_release().release();
```

Comment:
Oh, I don't think you can release, it would leak :/ I guess you need `auto os_release = find_os_release();` and then deref with `*` or `->` as needed. Or you can add a reference var if you want, but keeping the `unique_ptr` around and owning the `QFile` so that it gets destroyed when it goes out of scope.

Reply:
Memory leak 🤦🏽‍♂️, corrected.